src/tcache.c

```diff
@@ -306,8 +306,10 @@ tcache_gc_small(tsd_t *tsd, tcache_slow_t *tcache_slow, tcache_t *tcache,
 		if (tcache_slow->lg_fill_div[szind] > 1) {
 			tcache_slow->lg_fill_div[szind]--;
 		}
+		tcache_slow->lg_fill_div_offset[szind] = 0;
 		tcache_slow->bin_refilled[szind] = false;
 	}
+	assert(tcache_slow->lg_fill_div_offset[szind] == 0);
```

Comment:
How is this guaranteed? It seems we reset it either on flush, or right here 2 lines above. If it's indeed guaranteed, worth adding some comments explain why.

Since lg_fill_div and _offset are tied together, we should add helpers to handle both together, e.g. getter to return the effective lg_div, and helper to increment or decrement lg_fill_div which handles the lg_fill_div_offset internally. Then rename `lg_fill_div` itself to a longer name to make clear not access directly.

Reply:
It's similar to `bin_refilled`, when `low_water > 0`, there was no fill between GC, so the offset was not changed since last GC; otherwise when there was refill between GC, as inside the `else if (tcache_slow->bin_refilled[szind])` branch, we reset both `refill` to `false` and `lg_fill_div_offset` to 0, so it's again 0.